### DISCUSSION TITLE
Improve Elixir version detection

### DISCUSSION THREAD

**Discussion by henrythebuilder** [starship/starship#2572] [Ideas]
Hi, with current Starship (0.51.0) Elixir version is captured  as 'digit'. 
But there are three possible alternatives:
- _Elixir 1.11.4_
- _Elixir 1.12.0-RC.0_
- _Elixir 1.13.0-dev_

To detect the _full version_ I tried a new rule and may be enough to change the  `ELIXIR_VERSION_PATTERN` string in

```
const ELIXIR_VERSION_PATTERN: &str = "\
Erlang/OTP (?P<otp>\\d+)[^\\n]+

Elixir (?P<elixir>\\d[.\\d]+(-\\S*)?).*";
```

Sorry but at the moment I tried only the regular expression, I cannot test it in the project and make a new PR because I can't build the entire project due to many error during build (I don't know Rust)

Thanks
Enrico

------------------

Files changed :
```
diff --git a/src/modules/elixir.rs b/src/modules/elixir.rs
index 8d82365..67f46f2 100644
--- a/src/modules/elixir.rs
+++ b/src/modules/elixir.rs
@@ -9,7 +9,7 @@ use std::ops::Deref;
 const ELIXIR_VERSION_PATTERN: &str = "\
 Erlang/OTP (?P<otp>\\d+)[^\\n]+
 
-Elixir (?P<elixir>\\d[.\\d]+).*";
+Elixir (?P<elixir>\\d[.\\d]+(-\\S*)?).*";
 
 /// Create a module with the current Elixir version
 pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
```

**Comment by vladimyr** [comment]
@henrythebuilder Thanks for reporting! :+1: 

Could you paste the complete version output with rc or dev suffix so I can add an additional test to the Elixir module?

**Comment by henrythebuilder** [comment]
here the sample output:

```
Erlang/OTP 23 [erts-11.1.7] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1]

Elixir 1.11.3 (compiled with Erlang/OTP 21)

```

```
Erlang/OTP 23 [erts-11.1.7] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1]

Elixir 1.12.0-rc.0 (31d2b99) (compiled with Erlang/OTP 21)

```

```
Erlang/OTP 23 [erts-11.1.7] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1]

Elixir 1.13.0-dev (compiled with Erlang/OTP 23)
```

**Reply by vladimyr** [reply]
Thank you very much! :+1:

**Reply by henrythebuilder** [reply]
:+1:

**Comment by vladimyr** [comment]
PR created https://github.com/starship/starship/pull/2573

**Reply by henrythebuilder** [reply]
Thanks !
